discord/interaction.py

```diff
@@ -0,0 +1,661 @@
+            Controls the mentions being processed in this response.
+
+        Raises
+        --------
+        HTTPException
+            Sending the response failed.
+        NotFound
+            This interaction was not found.
+        Forbidden
+            The interaction token has expired.
+        InvalidArgument
+            You have already sent a response to this interaction or you specified
+            both ``embed`` and ``embeds`` or the length of ``embeds`` was invalid.
+
+        """
+        if self.responded:
+            raise InvalidArgument(f"Can not respond to {self} twice, use edit_response instead.")
+        if embeds is not None and embed is not None:
+            raise InvalidArgument('Cannot mix embed and embeds keyword arguments.')
+        if (embeds is not None) and len(embeds) > 10:
+            raise InvalidArgument('embeds has a maximum of 10 elements.')
+
+        if embed is not None:
+            embeds = [embed]
+
+        if self.is_ping():
+            response = InteractionResponse(pong=True)
+        else:
+            response = InteractionResponse(content=content, tts=tts, embeds=embeds,
+                                           allowed_mentions=allowed_mentions, ephemeral=ephemeral)
+        await self._state.http.create_interaction_response(self.id, self._token, response.to_dict())
+        self.responded = True
+
+    def edit_response(self, **fields):
+        """|coro|
+
+        Edit the response to this interaction.
+
+        The content must be able to be transformed into a string via ``str(content)``.
+
+        Parameters
+        ------------
+        content: Optional[:class:`str`]
+            The content to edit the response with or ``None`` to clear it.
+        embeds: List[:class:`Embed`]
+            A list of embeds to edit the response with.
+        embed: Optional[:class:`Embed`]
+            The embed to edit the response with. ``None`` suppresses the embeds.
+            This should not be mixed with the ``embeds`` parameter.
+        allowed_mentions: Optional[:class:`AllowedMentions`]
+            Controls the mentions being processed in the response. ``None``
+            suppresses all mentions.
+
+        Raises
+        -------
+        HTTPException
+            Editing the response failed.
+        Forbidden
+            The interaction token has expired.
+        InvalidArgument
+            You have not responded to this interaction yet or you specified
+            both ``embed`` and ``embeds`` or the length of ``embeds`` was invalid.
+        """
+        payload = {}
+        
+        if not self.responded:
+            raise InvalidArgument(f"Can not edit a response of {self} if no response was sent yet.")
+
+        try:
+            content = fields['content']
+        except KeyError:
+            pass
+        else:
+            if content is not None:
+                content = str(content)
+            payload['content'] = content
```

Comment:
```suggestion
        try:
            content = fields['content']
            if content is not None:
                content = str(content)
            payload['content'] = content
        except KeyError:
            pass
```
Although you might want to move the `payload[...] = ...` part into the if statement.

Reply:
Isn't it better to put the minimum amount of code inside of the `try` statement to be clear what error you're catching? At least I thought this was best practice